Update composer dependencies

jazzsequence opened the pull request:
* Removes lock files from version control
* Requires the consent API from [our forked version](https://github.com/humanmade/consent-api-js/)
* Removes the `class_exists` check for `WP_CONSENT_API` since that class won't exist and replaces it with a `defined` check for `WP_CONSENT_API_URL` which _should_ exist.

roborourke commented on composer.json:
```diff
@@ -3,7 +3,8 @@
     "require": {
-        "php": ">=7.1"
+        "php": ">=7.1",
+		"altis/consent-api": "^0.9"
```
Very minor thing, looks like the use of tabs vs spaces is inconsistent in this file. Altis modules themselves are using tabs.

jazzsequence replied:
👍 fixed

roborourke commented on .gitignore:
```diff
@@ -1,2 +1,4 @@
 /vendor/
 /node_modules/
+composer.lock
+package-lock.json
```
You don't necessarily need to remove the `package-lock.json` - any JS build would be specific to this repo still. It's only `composer.lock` files that are ignored in dependencies installed via composer

roborourke reviewed (commented):
In cases like this one thing you can do to allow the plugin to work as a standalone feature is to check for the existence of a local `vendor` folder and include the `vendor/autoload.php` if it exists. Then you get the best of both worlds.

jazzsequence commented:
@roborourke updated the minor issues. ready for a second look.

roborourke commented on inc/namespace.php:
```diff
@@ -16,6 +16,11 @@
  * Kick everything off.
  */
 function bootstrap() {
+	// If the consent api doesn't exist, load the local composer autoload file which should include it.
+	if ( ! class_exists( 'WP_CONSENT_API' ) ) {
+		require_once plugin_dir_path( __DIR__ ) . 'vendor/autoload.php';
```
This would probably work for some cases but is still not completely safe as it doesn't guarantee there is a local vendor directory. In addition the package type in its `composer.json` is `wordpress-plugin` - this means Composer will place it at `wp-content/plugins/..` by default. I think the best option is just to return early and trigger an error in this case and maybe show an admin notice.

If it has been added to the plugins directory then it's up to the site admin to activate it.

```suggestion
	// If the consent api doesn't exist, load the local composer autoload file which should include it.
	if ( ! class_exists( 'WP_CONSENT_API' ) ) {
		trigger_error( 'The WP Consent Level API plugin must be installed and activated', E_USER_WARNING );
		return;
```

roborourke reviewed (changes requested):
Small change to do - added a suggestion to show you what I mean.

jazzsequence commented:
added your suggested change @roborourke and also updated the `require` in the `composer.json` to point to the original package on packagist (now if only he could version releases...)

jazzsequence commented:
This will need one last update once https://github.com/humanmade/consent-api-js/pull/1 is merged or the JS API is merged into this repo.

jazzsequence commented:
@roborourke This should be good for another look. This just updates our dependencies to use our forked version of the JS API rather than the full plugin and PHP API.

roborourke reviewed (approved):
🚀
